kv/split_test.go

```diff
@@ -108,7 +108,7 @@ func TestRangeSplitMeta(t *testing.T) {
 	}
 
 	util.SucceedsSoon(t, func() error {
-		if _, _, err := engine.MVCCScan(context.Background(), s.Eng, keys.LocalMax, roachpb.KeyMax, 0, hlc.MaxTimestamp, true, nil); err != nil {
+		if _, _, err := engine.MVCCScan(context.Background(), s.Eng, keys.LocalMax, roachpb.KeyMax, engine.MaxSpanKeys, hlc.MaxTimestamp, true, nil); err != nil {
```

Comment:
I can only see parts of this diff (epic slow internet), but I assumed the special casing would go all the way - i.e. the `BatchRequest.Header` would have to contain `math.MaxInt64` (or something like that), and I further assumed that `engine.MaxSpanKeys` would be the corresponding constant to be used everywhere. Perhaps that isn't true. I'll take a closer look when I'm back in calmer waters.